common/git_worker.py

```diff
@@ -238,6 +238,7 @@ def extract_all_repos(self):
 
         for repo in self.repo_states:
             if not repo.commit_id:
+                repo.commit_id = 'head'
```

Comment:
It's not quite obvious for me to understand why we set 'head' to commit_id here. 

It's better to update constructor of ProductState class and modify 'if' conditions in extract_all_repos method

Reply:
That is good idea, I'll try to test it.